libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1039,6 +1041,16 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         CacheHandleWrapper.requestRoadGraphDataUpdate(navigator.cache, callback)
     }
 
+    /**
+     * Returns leg index the user is currently on. If no RouteProgress updates are available,
+     * returns the value passed as `initialLegIndex` parameter to `setNavigationRoutes`.
+     * If no routes are set, returns 0.
+     */
+    fun currentLegIndex(): Int {
```

Comment:
What do you think about nullable value?
```suggestion
     * If no routes are set, returns null.
     */
    fun currentLegIndex(): Int? {
```
plus make the route line accept null and fallback to 0, or turn off leg highlighting if no leg data is present?

Reply:
> plus make the route line accept null and fallback to 0, or turn off leg highlighting if no leg data is present?

Not so sure. Consider the case of route preview. Routes have not yet been set to MapboxNavigation, but they are set to `MapboxRouteLineApi`. So I think it's a valid case that no leg data is present when the method in invoked. And I don't think that we should turn off leg highlighting in this case.